usage/rest-server/src/test/java/brooklyn/rest/resources/ServerResourceTest.java

```diff
@@ -97,5 +103,38 @@ public void testShutdown() throws Exception {
                 assertFalse(getManagementContext().isRunning());
             }});
     }
-    
+
+    @Test
+    void testGetConfig() throws Exception {
+        ((ManagementContextInternal)getManagementContext()).getBrooklynProperties().put("foo.bar.baz", "quux");
+        try {
+            assertEquals(client().resource("/v1/server/config/foo.bar.baz").get(String.class), "quux");
+        } finally {
+            ((ManagementContextInternal)getManagementContext()).getBrooklynProperties().remove("foo.bar.baz");
+        }
+    }
+
+    @Test
+    void testGetMissingConfigThrowsException() throws Exception {
+        final String key = "foo.bar.baz";
+        BrooklynProperties properties = ((ManagementContextInternal)getManagementContext()).getBrooklynProperties();
+        Object existingValue = null;
+        boolean keyAlreadyPresent = false;
+        String response = null;
+        if (properties.containsKey(key)) {
+            existingValue = properties.remove(key);
+            keyAlreadyPresent = true;
+        }
+        try {
+            response = client().resource("/v1/server/config/" + key).get(String.class);
+        } catch (UniformInterfaceException e) {
+            assertEquals(e.getResponse().getStatus(), 204);
```

Comment:
Does this test pass?! It's asserting a 204 rather than a 404 (i.e. "No Content').

My personal preference is to put the `Asserts.fail` inside the try block, and not to do a `return` inside the catch block. I find that clearer, but no strong feelings.